File: algorithm/sort/insertion_sort.py
# reverse = True : sort from greatest to least
def insertion_sort(arr, reverse=False):
    for i in range(1, len(arr)):
        for j in range(i, 0, -1):
            state = arr[j] > arr[j-1] if reverse else arr[j] < arr[j-1]
            if state:
                arr[j], arr[j-1] = arr[j-1], arr[j]
            else: break
                
    return arr

File: algorithm/sort/insertion sort/insertion_sort.cpp
// return a pointer to arr
int* insertion_sort(int arr[], int arr_len, bool reverse = false){
    int temp;
    bool state;
    for (int i = 1; i < arr_len; i++){
        for (int j = i; j > 0; j--){
            state = (reverse == true)? arr[j] > arr[j-1] : arr[j] < arr[j-1];

            if (state){
                temp = arr[j-1];
                arr[j-1] = arr[j];
                arr[j] = temp;

            } else{
                break;
            }
        }
    }
    return arr;
}
